src/core.c/CompUnit/Repository/Installation.rakumod

```diff
@@ -744,6 +816,239 @@ Please ask the author to add a \"provides\" section, mapping every exposed names
+        }
+
+        sub process-path(Str() $path) {
+            if $path ~~ /
+                    "<" ( "plain" || "cmd-args" || "plat-sep" || "abs" || "abs-slash" ) ">"
+                    (.*)
+                / {
+                my $op = $0.Str;
+                my $val = $1.Str;
+                given $op {
+                    when "plain" | "plat-sep" {
+                        return '"' ~ sh-escape($val) ~ '"';
+                    }
+                    when "cmd-args" {
+                        return '"$@"';
+                    }
+                    when "abs" | "abs-slash" {
+                        return '"${DIR}/' ~ sh-escape($val) ~ '"';
+                    }
+                }
+            }
+            else {
+                note "Don't understand this arg: $path";
+                exit 1;
+            }
+        }
+
+        # Write program
+        unless $program ~~ / ^ "<" ( "plain" || "plat-sep" || "abs" ) ">" / {
+            note "Program is invalid. Must start with \"<plain>\", \"<plat-sep>\", \"<abs>\".";
+            exit 1;
+        }
+        @call-parts.push: process-path($program);
+
+        # Write args
+        for @args -> $arg {
+            @call-parts.push: process-path($arg);
+        }
+
+        my @lines;
+
+        # Write CWD
+        if $cwd {
+            unless $cwd ~~ / ^ "<" ( "plain" || "abs" ) ">" / {
+                note "CWD is invalid. Must start with \"<plain>\" or \"<abs>\".";
+                exit 1;
+            }
+            @lines.push: 'cd ' ~ process-path($cwd);
+        }
+
+        # Write env vars
+        for %env-add.keys.sort -> $key {
+            validate-env-var $key;
+            my $val = %env-add{$key};
+            @lines.push: "export $key=$val"
+        }
+
+        @env-remove.map: -> $name {
+            validate-env-var $name;
+            @lines.push: "unset $name"
+        }
+
+        @lines.push: @call-parts.join(" ");
+
+        
+        my $text = get-template('posix-runner-tmpl').slurp;
+        $text ~= @lines.join("\n");
+        $text
+    }
+
+    sub get-template($name) {
+        nqp::gethllsym('default', 'SysConfig').rakudo-home.IO.add("templates").add($name)
+    }
+
+    sub create-config-blob(
+        Str           :$program!,
+        Str           :$arg0,
+        Str           :$cwd,
+                      :@args,
+                      :%env-add,
+                      :@env-remove,
+    ) {
+        my $char-size = Rakudo::Internals.IS-WIN ?? 2 !! 1;
+        my $encoding = Rakudo::Internals.IS-WIN ?? 'utf16le' !! 'utf8';
+        my @end-str-bytes = Rakudo::Internals.IS-WIN ?? (0x00, 0x00) !! (0x00,);
+
+        my $marker = "EXEC_RUNNER_WRAPPER_CONFIG_MARKER".encode($encoding);
+
+        my Buf $cbuf .= new;
+        my Int $pos = 0;
+
+        sub write-str(Str() $str) {
+            my $bytes = $str.encode($encoding);
+            $cbuf.write-uint16: $pos, ($bytes.elems / $char-size + 1).Int;
```

Comment:
I think this should throw if $bytes.elems exceeds 2**16

Reply:
Done.